### PR TITLE
Make all linters happy.

### LINKED ISSUE #1038: code: Remove `pylint:disable=...` comments
[Now](#1033) that we use ruff we should remove all pylint disable comments.
When doing this we could also run `ruff --fix` for a selected amount of rules.

Also we could do everything that is here: https://beta.ruff.rs/docs/configuration/#automatic-noqa-management

### PR DESCRIPTION
**Pull request by maringuu**
Now that we switched to ruff we can easily make all linters happy on all files.

While this adds a lot (about 700 of noqa comments) this is arguably better than having the linter fail.
With the current state I developed the bad habit to just do `git commit --no-verify` because:
* pylint is slow (fixed by using ruff)
* The linter would fail anyways (because we had so many linting errors in the codebase)

Now after this PR this bad habit is not needed anymore and the linters will only produce true positives for changes I made.

Closes #1038 #894

### CODE REVIEW

File: src/objects/firmware.py [hunk 1]
```diff
@@ -78,16 +78,16 @@ def __init__(self, **kwargs):
...
         #:
-        #: This attribute is **optional**. The release date is assumed to be the start of UNIX epoch time (`1970-01-01`) if not specificed.
+        #: This attribute is **optional**. The release date is assumed to be the start of UNIX epoch time (`1970-01-01`) if not specificed.  # noqa: E501
         self.release_date: str | None = None
 
-        #: User-defined firmware tags for advanced grouping and filtering of firmware images, saved as {'tag': :class:`helperFunctions.tag.TagColor`} dictionary.
-        #: It is important to understand that these tags are **separately stored** from the :attr:`objects.file.FileObject.analysis_tags`, which are propagated by analysis plugins.
+        #: User-defined firmware tags for advanced grouping and filtering of firmware images, saved as {'tag': :class:`helperFunctions.tag.TagColor`} dictionary.  # noqa: E501
+        #: It is important to understand that these tags are **separately stored** from the :attr:`objects.file.FileObject.analysis_tags`, which are propagated by analysis plugins.  # noqa: E501
```
Comment (maringuu) [inline src/objects/firmware.py:90]: If we want to fix errors in this PR then these would be low hanging fruits.

File: src/plugins/analysis/architecture_detection/install.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3  # noqa: EXE001
```
Comment (maringuu) [inline src/plugins/analysis/architecture_detection/install.py:1]: This also.

**Review by maringuu — COMMENTED** [review]
Tbh. I really would not like to do any fixes in this PR.
I'd be happy to create a list of errors that we don't want to have in the codebase and should not be ignored.
We can then run just `s/  # noqa: .*E501.*$//` on the whole codebase again to remove the noqa comments and then ruff will complain again and we can fix them.

Things I'd like to not have in the codebase (and are trivial to fix):
- E501
- B007
- EXE001
- ARG

Rule list: https://beta.ruff.rs/docs/rules/

**Comment by jstucke** [comment]
It's probably more efficient to run ruff again instead of trying to fix the 9999 conflicts

**Comment by maringuu** [comment]
> 9999

Ofc!

File: src/plugins/analysis/device_tree/internal/device_tree_utils.py [hunk 3]
```diff
@@ -72,13 +72,15 @@ def header_has_illegal_values(header: DeviceTreeHeader, max_size: int) -> bool:
         header.struct_block_size,
         header.strings_block_size,
     ]
-    return header.version > 20 or any(n > max_size or n > header.size for n in values) or header.size > max_size
+    return (
+        header.version > 20  # noqa: PLR2004
+        or any(n > max_size or n > header.size for n in values)
+        or header.size > max_size  # noqa: PLR2004, RUF100
+    )  # noqa: PLR2004, RUF100
```
Comment (jstucke) [inline src/plugins/analysis/device_tree/internal/device_tree_utils.py:79]: ...
black vs ruff 
1. ruff adds "# noqa: PLR2004"
2. black breaks the line
3. the "PLR2004" doesn't fit anymore so ruff adds a "RUF100" and a new "PLR2004"
4. the line is now even longer so black breaks it again
5. the second "PLR2004" doesn't fit anymore so ruff adds another "RUF100" and also a new "PLR2004"

File: src/plugins/analysis/known_vulnerabilities/test/test_known_vulnerabilities.py [hunk 4]
```diff
@@ -51,39 +55,41 @@ def test_process_object_software_wrong_version(self, analysis_plugin):
...
             'result': {'sha256': '7579d10e812905e134cf91ad8eef7b08f87f6f8c8e004ebefa441781fea0ec4a'}
         }
         test_file.processed_analysis['software_components'] = {}
 
         results = analysis_plugin.process_object(test_file).processed_analysis[analysis_plugin.NAME]
 
-        assert len(results) == 3, 'incorrect number of vulnerabilities found (summary + tag + one result)'
+        assert (
+            len(results) == 3  # noqa: PLR2004
+        ), 'incorrect number of vulnerabilities found (summary + tag + one result)'  # noqa: PLR2004, RUF100
```
Comment (jstucke) [inline src/plugins/analysis/known_vulnerabilities/test/test_known_vulnerabilities.py:68]: see above

File: src/plugins/analysis/users_and_passwords/internal/update_password_list.py [hunk 5]
```diff
@@ -22,8 +22,12 @@ def _setup_argparser():
...
     parser.add_argument('-V', '--version', action='version', version=f'{PROGRAM_NAME} {PROGRAM_VERSION}')
     parser.add_argument('-d', '--debug', action='store_true', default=False, help='print debug messages')
-    parser.add_argument('-p', '--password_lists_directory', default=os.path.join(THIS_FILE_DIR, 'passwords'))
-    parser.add_argument('-o', '--output_file', default=os.path.join(THIS_FILE_DIR, '../../../..', 'bin/passwords.txt'))
+    parser.add_argument(
+        '-p', '--password_lists_directory', default=os.path.join(THIS_FILE_DIR, 'passwords')  # noqa: PTH118
+    )  # noqa: PTH118, RUF100
+    parser.add_argument(
+        '-o', '--output_file', default=os.path.join(THIS_FILE_DIR, '../../../..', 'bin/passwords.txt')  # noqa: PTH118
+    )  # noqa: PTH118, RUF100
```
Comment (jstucke) [inline src/plugins/analysis/users_and_passwords/internal/update_password_list.py:30]: see above

**Review by jstucke — APPROVED** [review]

**Pull request merged**